util/design/gen-otp-img.py

```diff
@@ -0,0 +1,183 @@
+#!/usr/bin/env python3
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+r"""Generate RTL and documentation collateral from OTP memory
+map definition file (hjson).
+"""
+import argparse
+import datetime
+import logging as log
+import random
+
+import hjson
+from lib.common import wrapped_docstring
+from lib.OtpMemImg import OtpMemImg
+
+# Get the memory map definition.
+MMAP_DEFINITION_FILE = 'hw/ip/otp_ctrl/data/otp_ctrl_mmap.hjson'
+# Life cycle state and ECC poly definitions.
+LC_STATE_DEFINITION_FILE = 'hw/ip/lc_ctrl/data/lc_ctrl_state.hjson'
+# Default image file definition (can be overridden on the command line).
+IMAGE_DEFINITION_FILE = 'hw/ip/otp_ctrl/data/otp_ctrl_img_dev.hjson'
+# Default output path (can be overridden on the command line).
+MEMORY_HEXFILE = 'otp-img.mem'
+
+
+def _override_seed(args, name, config):
+    '''Override the seed key in config with value specified in args'''
+    arg_seed = getattr(args, name)
+    if arg_seed:
+        log.warning('Commandline override of {} with {}.'.format(
+            name, arg_seed))
+        config['seed'] = arg_seed
+    # Otherwise, we either take it from the .hjson if present, or
+    # randomly generate a new seed if not.
+    else:
+        random.seed()
+        new_seed = random.getrandbits(64)
+        if config.setdefault('seed', new_seed) == new_seed:
+            log.warning('No {} specified, setting to {}.'.format(
+                name, new_seed))
+
+
+def main():
+    log.basicConfig(level=log.INFO, format="%(levelname)s: %(message)s")
+
+    parser = argparse.ArgumentParser(
+        prog="gen-otp-img",
+        description=wrapped_docstring(),
+        formatter_class=argparse.RawDescriptionHelpFormatter)
+    parser.add_argument('--img-seed',
+                        type=int,
+                        metavar='<seed>',
+                        help='''
+                        Custom seed for RNG to compute randomized items in OTP image.
+
+                        Can be used to override the seed value specified in the image
+                        config Hjson.
+                        ''')
+    parser.add_argument('--lc-seed',
+                        type=int,
+                        metavar='<seed>',
+                        help='''
+                        Custom seed for RNG to compute randomized life cycle netlist constants.
+
+                        Note that this seed must coincide with the seed used for enerating
+                        the LC state encoding (gen-lc-state-enc.py).
+
+                        This value typically does not need to be specified as it is taken from
+                        the LC state encoding definition Hjson.
```

Comment:
Does it ever need to be specified? If not, we could remove the argument... Similarly for `--otp-seed`.

Reply:
currently we do not use them, but I've provisioned these switches since we may need them at some point for running our nightly regressions with different netlist constant seeds.